include/oneapi/dpl/pstl/parallel_backend_utils.h

```diff
@@ -186,7 +188,14 @@ __set_intersection_construct(_ForwardIterator1 __first1, _ForwardIterator1 __las
         {
             if (!__comp(*__first2, *__first1))
             {
-                ::new (::std::addressof(*__result)) _Tp(*__first1);
+
+                auto __select_element = [](auto&& _f1, auto&& _f2) {
```

Comment:
As far as I understood this code, `_f1' and `_f2` are iterators. So why we using `std::forward` here?
Moreover, may be `__select_element` lambda not required at all?

Reply:
What about `_copy(*(_CopyFromFirstSet::value ? _f1 : _f2), *__result);`